contracts/multisig-prover/src/contract.rs

```diff
@@ -1,44 +1,479 @@
 #[cfg(not(feature = "library"))]
 use cosmwasm_std::entry_point;
-use cosmwasm_std::{to_binary, Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
+use cosmwasm_std::{
+    from_binary, to_binary, wasm_execute, Binary, BlockInfo, Deps, DepsMut, Env, HexBinary,
+    MessageInfo, QuerierWrapper, QueryRequest, Reply, Response, StdError, StdResult, SubMsg,
+    WasmMsg, WasmQuery,
+};
+use cw_utils::{parse_reply_execute_data, MsgExecuteContractResponse};
+
+use std::{collections::HashMap, str::FromStr};
+
+use axelar_wasm_std::{Participant, Snapshot};
+use connection_router::{msg::Message, types::ChainName};
+use multisig::{msg::Multisig, types::MultisigState};
+use service_registry::state::Worker;
 
 use crate::{
     error::ContractError,
+    events::Event,
     msg::ExecuteMsg,
-    msg::{GetProofResponse, QueryMsg},
+    msg::{GetProofResponse, InstantiateMsg, ProofStatus, QueryMsg},
+    state::{Config, COMMANDS_BATCH, CONFIG, KEY_ID, PROOF_BATCH_MULTISIG, REPLY_BATCH},
+    types::{BatchID, CommandBatch, ProofID},
 };
 
+const START_MULTISIG_REPLY_ID: u64 = 1;
+
 #[cfg_attr(not(feature = "library"), entry_point)]
-pub fn execute(
-    _deps: DepsMut,
+pub fn instantiate(
```

Comment:
It's too hard to review it when `instantiate` and all execute methods are done in one single PR. Please split in the future. One method at a time.